kms/src/main/java/io/kroxylicious/kms/service/Kms.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.service;
+
+import java.util.concurrent.CompletionStage;
+
+import javax.crypto.SecretKey;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Abstracts the KMS operations needed for Envelope Encryption
```

Comment:
Nice job on this interface.

Reply:
TBH this feels like the easy bit. I'm still not completely happy with the `DekContext`...